### PR TITLE
Add tests for GuardedResources: Workspace, WorkspaceInstance and WorkspaceLog (1/2)

### PR DESCRIPTION
**Pull request by geropl**
## Description

This PR adds tests for workspace-like GuardedResources: `Workspace`, `WorkspaceInstance` and `WorkspaceLog`.
It's a precursor for a follow-up PR.

## Related Issue(s)

Context: https://github.com/gitpod-io/gitpod/pull/10696

## How to test
 - `cd components/server && yarn test`

## Release Notes

```release-note
NONE
```

## Werft options:

- [ ] /werft with-preview

### CODE REVIEW

File: components/server/src/workspace/gitpod-server-impl.ts [hunk 1]
```diff
@@ -1664,7 +1664,7 @@ export class GitpodServerImpl implements GitpodServerWithTracing, Disposable {
         const wsiPromise = this.workspaceDb.trace(ctx).findInstanceById(instanceId);
         const teamMembers = await this.getTeamMembersByProject(ws.projectId);
 
-        await this.guardAccess({ kind: "workspace", subject: ws, teamMembers }, "get");
+        await this.guardAccess({ kind: "workspaceLog", subject: ws, teamMembers }, "get");
```
Comment (geropl) [inline components/server/src/workspace/gitpod-server-impl.ts:1667]: For workspace logs we have a separate resource kind "workspaceLog".

**Comment by jankeromnes** [comment]
> This PR adds tests for workspace-like GuardedResources: `Workspace`, `WorkspaceInstance` and `WorkspaceLog`.
It's a precursor for a follow-up PR.

Hey, that's awesome, many thanks for adding tests @geropl! 🥇 💯

> ## How to test

If the tests already run as part of the build, I guess this could be simply: "1. Should build:"

> ## Release Notes
> ```
> added tests for GuardedResources Workspace, WorkspaceInstance and WorkspaceLog
> ```

FYI, I think this should rather be `NONE`, because we've decided to make release notes more customer-centric and less technical-detail-oriented. 💡

Source: [Slack announcement](https://gitpod.slack.com/archives/C01KGM9AW4W/p1654677957766019) (internal)

**Comment by geropl** [comment]
> If the tests already run as part of the build, I guess this could be simply: "1. Should build:"

Yes, or `(cd components/server && yarn test)`. :+1: 

> FYI, I think this should rather be NONE, because we've decided to make release notes more customer-centric and less technical-detail-oriented. bulb

Thx, was not aware - adjusted.

File: components/server/src/auth/resource-access.ts [hunk 2]
```diff
@@ -463,16 +463,26 @@ export class RepositoryResourceGuard implements ResourceAccessGuard {
...
+        let workspace: Workspace;
+        switch (resource.kind) {
+            case "workspaceLog":
+                workspace = resource.subject;
+                break;
+            case "snapshot":
+                workspace = resource.workspace;
+                break;
+            default:
+                // We do not handle resource kinds here!
```
Comment (jankeromnes) [inline components/server/src/auth/resource-access.ts:481]:
```suggestion
                // We do not handle other resource kinds here!
```

**Review by jankeromnes — APPROVED** [review]
Many thanks!

Code looks good to me, especially if it does what we expect it to. 👍

Holding because of two (optional) open questions:

- Shouldn't we run a new build with-preview to give the new guards a quick sanity check?
- I see in the build logs that `workspaceLikeResourceGuardsCanAcccess` ran and did pass ✅ however, out of abundance of caution, did you observe actual failures? (E.g. while developing, or by flipping one random expected boolean to see if the test actually fails)

/hold

**Comment by jankeromnes** [comment]
Lol, and my hold? 😆

**Pull request merged**